flocker/node/test/test_script.py

```diff
@@ -304,180 +469,170 @@ def test_custom_config_file(self):
     return Tests
 
 
-class AgentConfigFromFileTests(SynchronousTestCase):
+class ValidateConfigurationTests(SynchronousTestCase):
     """
-    Tests for :func:`agent_config_from_file`.
+    Tests for :func:`validate_configuration`.
     """
 
     def setUp(self):
-        self.scratch_directory = FilePath(self.mktemp())
-        self.scratch_directory.makedirs()
-        self.config = self.scratch_directory.child('config.yml')
+        # This is a sample working configuration which tests can modify.
+        self.configuration = {
+            u"control-service": {
+                u"hostname": u"10.0.0.1",
+                u"port": 1234,
+            },
+            u"dataset": {
+                u"backend": u"zfs",
+                u"zfs-pool": u"custom-pool",
+            },
+            "version": 1,
+        }
 
-    def assertErrorForConfig(self, exception, message, configuration=None):
+    def test_valid_zfs_configuration(self):
+        """
+        No exception is raised when validating a valid configuration with a
+        ZFS backend.
         """
-        Assert that given a particular configuration,
-        :func:`agent_config_from_file` will fail with an expected exception
-        and message.
+        # Nothing is raised
+        validate_configuration(self.configuration)
 
-        :param Exception exception: The exception type which
-            :func:`agent_config_from_file` should fail with.
-        :param dict configuration: The contents of the agent configuration
-            file. If ``None`` then the file will not exist.
-        :param bytes message: The expected exception message.
+    def test_valid_loopback_configuration(self):
+        """
+        No exception is raised when validating a valid configuration with a
+        ZFS backend.
         """
-        if configuration is not None:
-            self.config.setContent(yaml.safe_dump(configuration))
+        self.configuration['dataset'] = {
+            u"backend": u"loopback",
+            u"loopback-pool": u"custom-pool",
+        }
```

Comment:
I'll try it and then decide what I prefer if that's OK. `build_schema_test` seems more suited to schemas which are one and maybe two lines long, it might get unwieldy.